src/main/java/org/usfirst/frc/team4099/robot/subsystems/Arm.kt

```diff
@@ -131,46 +141,53 @@ class Arm private constructor() : Subsystem {
                     return
                 }
                 //if current time - init time > 0.5 : skip loop: set to stationary
-
-                when (armState) {
-                    Arm.ArmState.LOW -> {
-                        setArmPosition(ArmState.LOW.targetPos)
-                    }
-                    Arm.ArmState.EXCHANGE -> {
-                        setArmPosition(ArmState.EXCHANGE.targetPos)
-                    }
-                    Arm.ArmState.HIGH -> {
-                        setArmPosition(ArmState.HIGH.targetPos)
+                if (movementState == Arm.MovementState.STATIONARY) {
+                    brake.set(DoubleSolenoid.Value.kReverse)
+                } else if (movementState == Arm.MovementState.HOLD) {
+                    masterSRX.set(ControlMode.MotionMagic, ArmConversion.pulsesToRadians(masterSRX.sensorCollection.pulseWidthPosition))
+                } else {
+                    brake.set(DoubleSolenoid.Value.kForward)
+                    when (armState) {
```

Comment:
On second thought, this would probably be better off as part of the `if` statement - `LOW`, `EXCHANGE`, and `HIGH` all do the same thing, `STILL` has its own action, and `VELOCITY_CONTROL` doesn't need to be handled.